clang/lib/Parse/ParseOpenACC.cpp

```diff
@@ -298,7 +367,18 @@ void Parser::ParseOpenACCDirective() {
         T.consumeClose();
       break;
     }
+    case OpenACCDirectiveKind::Cache:
+      ParseOpenACCCacheVarList();
+      // The ParseOpenACCCacheVarList function manages to recover from failures,
+      // so we can always consume the close.
+      T.consumeClose();
+      break;
     }
+  } else if (DirKind == OpenACCDirectiveKind::Cache) {
+    // Cache's paren var-list is required, so error here if it isn't provided.
+    // We know that the consumeOpen above left the first non-paren here, so use
+    // expectAndConsume to emit the proper dialog, then continue.
+    (void)T.expectAndConsume();
```

Comment:
Why do not call for ParseOpenACCCacheVarList(); here?

Reply:
I was considering the lack of a `(` to mean "the author provided nothing and continued with the clause-lists", so it didn't seem to make sense to try to parse a var list.  

Do you consider treating it as if it is JUST a missed '(' to make more sense?

Reply:
Ah, ok, I see. Yes, I think better explicitly diagnose missed '(' here